src/error.rs

```diff
@@ -67,14 +64,12 @@ pub enum Error {
     TlsError { message: String },
 }
 
-#[cfg(any(
-    feature = "mysql-16",
-    feature = "postgresql-0_16",
-    feature = "rusqlite-0_19"
-))]
-impl From<r2d2::Error> for Error {
-    fn from(e: r2d2::Error) -> Error {
-        Error::ConnectionError(e.into())
+impl From<mobc::Error<failure::Compat<Error>>> for Error {
+    fn from(e: mobc::Error<failure::Compat<Error>>) -> Self {
```

Comment:
It's going to get nicer when we switch away from failure. Maybe it's something we should consider doing before 0.1?

Reply:
Naturally.